src/CommunitySleep.js

```diff
@@ -0,0 +1,102 @@
+    const userSleep = this.sleeps.filter(sleeper => (sleeper.userID === userID));
+    const amountSlept = userSleep.map(sleeper => sleeper.hoursSlept);
+    const avgHoursSlept = (amountSlept.reduce((a, b) => a + b, 0)) / amountSlept.length;
+    return Math.round(avgHoursSlept * 10) / 10;
+  }
+
+  calculateAvgSleepQualPerDay(userID) {
+    const userSleep = this.sleeps.filter(sleeper => (sleeper.userID === userID));
+    const sleepQuality = userSleep.map(sleeper => sleeper.sleepQuality);
+    const avgSleepQuality = (sleepQuality.reduce((a, b) => a + b, 0)) / sleepQuality.length;
+    return Math.round(avgSleepQuality * 10) / 10;
+  }
+
+  findHrsSleptOnDay(userID, date) {
+    const hoursSlept = this.sleeps.find(sleeper => sleeper.userID === userID && sleeper.date === date);
+    return hoursSlept.hoursSlept;
+  }
+
+  findSleepQualityOnDay(userID, date) {
+    const sleepQuality = this.sleeps.find(sleeper => sleeper.userID === userID && sleeper.date === date);
+    return sleepQuality.sleepQuality;
+  }
+
+  calculateSleepStatsWeek(userID, startDate, endDate) {
+    const sleepWeekTotals = [];
+    const startDateNumber = this.convertDateString(startDate);
+    const endDateNumber = this.convertDateString(endDate);
+    const userSleep = this.findUserSleepData(userID);
+    const sleepDates = userSleep.filter(sleeper => {
+      const sleepDateToNumber = this.convertDateString(sleeper.date)
+      if(sleepDateToNumber >= startDateNumber && sleepDateToNumber <= endDateNumber){
+        sleepWeekTotals.push(sleeper);
+      }
+    })
+    return sleepWeekTotals;
+  }
+
+  calculateAvgSleepQualityWk(userID, startDate, endDate) {
+    const something = this.calculateSleepStatsWeek(userID, startDate, endDate);
+    const sleepQualityTotals = something.map(sleeper => sleeper.sleepQuality)
+    const sleepWeekAvg = (sleepQualityTotals.reduce((a, b) => a + b, 0))/7
+    return Math.round(sleepWeekAvg * 10) / 10
+  }
+
+  calculateAvgSleepQuality() {
+    const sleepQualityAllUsers = this.sleeps.map(allSleepers => allSleepers.sleepQuality);
+     const avgSleepQuality = (sleepQualityAllUsers.reduce((a, b) => a + b, 0)) / sleepQualityAllUsers.length;
+     return Math.round(avgSleepQuality * 10) / 10;
+  }
+
+  findBestQualitySleepers(startDate, endDate) {
+    const bestSleepers = []
+    const startDateNumber = this.convertDateString(startDate);
+    const endDateNumber = this.convertDateString(endDate);
+    const sleepWeek = this.sleeps.filter(sleeper => {
+      const sleepDateToNumber = this.convertDateString(sleeper.date)
+      if (sleepDateToNumber >= startDateNumber && sleepDateToNumber <= endDateNumber) {
+        return sleeper;
+      }
+    })
+    const filterSleepers = sleepWeek.filter(sleeper => {
+      const avgSleepQual = this.calculateAvgSleepQualityWk(sleeper.userID, startDate, endDate)
+      if(!bestSleepers.includes(sleeper.userID) && avgSleepQual > 3){
+        bestSleepers.push(sleeper.userID)
+      }
+    })
```

Comment:
@srslie then why did it work???? I'm not sure what you mean by "I would either change this into a just the return of that condition".

Reply:
It works because it's an iterator method, and all these methods basically work by being 'for loops' that go through the array and have some special properties. Filters are set up sorta like:
```
let elementsThatPassFilterCondition = [ ]
for (var i = 0; i < array.length; i++) {
    if (array[i] === condition) {
       elementsThatPassFilterCondition.push(array[i]
    }
}
```
So, you're just coding in some code that's already in there because of the filter. Totally fine to refactor later, but just a note for using that method.